src/macports1.0/macports.tcl

```diff
@@ -1135,7 +1135,7 @@ match macports.conf.default."
                     set macports::build_arch x86_64
                 }
             } elseif {$os_major >= 10} {
-                if {[sysctl hw.cpu64bit_capable] == 1} {
+                if {$tcl_platform(machine) eq "x86_64"} {
```

Comment:
This is not correct on 64-bit Macs that use the 32-bit kernel.
```
$ sw_vers -productVersion; echo 'puts $tcl_platform(machine)' | tclsh; sysctl hw.cpu64bit_capable
10.6.8
i386
hw.cpu64bit_capable: 1
```

Reply:
Ah good catch. That's pretty cool, but also makes things a bit tricky. Let me have a think about this.

Would you mind running ` getconf LONG_BIT`? Also, what is the processor?

Reply:
The processor is irrelevant. Apple decides which 64-bit Macs use the 32-bit kernel by default and which use the 64-bit kernel by default, and it can vary by OS version, and the user can [override the default](https://web.archive.org/web/20150323044856/https://support.apple.com/en-us/HT3773).

On this 64-bit 10.6 VM, `getconf LONG_BIT` says `64`, and on a 32-bit 10.6 VM, it says `32`, so that may indeed be a useful replacement for `sysctl hw.cpu64bit_capable`.

(I also checked on a PowerBook G4 (which is 32-bit) and under both 10.4 and 10.5 `getconf LONG_BIT` says `32`. I did not test on a Power Mac G5 (which is 64-bit); presumably it says `64` there. The current code was only running `sysctl hw.cpu64bit_capable` for 10.6 and later anyway so I guess this is irrelevant.)

Reply:
> On this 64-bit 10.6 VM, getconf LONG_BIT says 64, and on a 32-bit 10.6 VM, it says 32, so that may indeed be a useful replacement for sysctl hw.cpu64bit_capable.

Great. I've updated it to use getconf. Thanks for testing @ryandesign.